magento/pwa-studio
# [PWA-2590] A component library in Peregrine

## Description

Since our UI components are tightly coupled with logic (talons) that live in Peregrine, we would like Peregrine to once again become the home for UI components. It doesn't make much sense to publish two separate packages that are so tightly coupled, except for perhaps from a bundle size perspective, and that is a solvable problem.

All of our future demo storefronts (and merchant storefronts) would get their componentry from this one place. This would also be the package they target for extensions.

## Acceptance Criteria
- Peregrine publishes a new "ui" directory as a sibling of "hooks" and "talons"
- All existing Peregrine UI components move into this new "ui" directory
  - ContainerChild
  - List
  - Price
- All existing Venia UI components move into this new "ui" directory
- The `venia-ui` package is removed from PWA Studio and no longer gets updated during releases
- All UI component references in other components and storefronts are updated to import from "@magento/peregrine/lib/ui"
- Storefronts no longer list "venia-ui" in their dependencies
- Peregrine is outfitted such that only the components needed by a storefront are bundled. We do not want to bundle all components all the time.

## Related Issue







* [[PWA-2590](https://jira.corp.magento.com/browse/PWA-2590)]

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature

##### Test Plan
1. Verify venia-ui package is removed. (Move this scenario to respective story if not done part of this story).
2. Run Cypress all tests and make sure everything works.
3. Run yarn install, build, stage, watch, bundle size, build:Report cli commands still work.
4. Create scaffolding and verify all supported cli commands still work.
5. Quickly verify UI in Firefox, Safari , Android chrome and iOS safari.
6. Verify cypress data attributes are still controllable via .env > BABEL_KEEP_ATTRIBUTES=true variable.
7. Verify lighthouse and webpage test scores.

##### Test Coverage
1. No UI test cases are needed but all new changes should be covered in Unit test cases.

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
